devauth/devauth.go

```diff
@@ -519,22 +557,27 @@ func (d *DevAuth) updateDeviceStatus(ctx context.Context, devId, status string,
 		return errors.Wrap(err, "failed to update device status")
 	}
 
-	b, err := json.Marshal([]string{devId})
-	if err != nil {
-		return errors.New("internal error: cannot marshal array into json")
-	}
 	// submit device status change job
 	if statusChanged {
+		dev, err := d.db.GetDeviceById(ctx, devId)
```

Comment:
This extra GetDeviceById can probably be avoided using FindAndModify in SetDeviceIdentity.
It doesn't matter, as we do not expect to change statuses so often, let's keep it as it is.

Reply:
you mean in the UpdateDevice, right?
I'll follow up on this, we also call GetDeviceStatus which is not always needed